### ISSUE TITLE
Add scrollBehavior prop

### ISSUE THREAD

**Issue by olivier-drieux** [richardscarrott/react-snap-carousel#18]
By default, the scroll behavior in the hook is 'smooth' (see [use-snap-carousel.tsx:169](https://github.com/richardscarrott/react-snap-carousel/blob/main/src/use-snap-carousel.tsx#L169)).
<img width="644" alt="Capture d’écran 2023-08-25 à 10 11 28" src="https://github.com/richardscarrott/react-snap-carousel/assets/27553827/976ff3c9-22f8-4229-ac14-71a58f8aef44">
There is a known bug on chromium based browsers with smooth scroll behavior (ex: https://bugs.chromium.org/p/chromium/issues/detail?id=1299237).
I would like to be able to change the scroll behavior when using the useSnapCarousel hook.
I can do a PR if you want to.

**Comment by richardscarrott** [comment]
Can't say I've run into that issue myself but happy to accept a PR.

I think it'd make most sense to pass it into `goTo`, `next` and `prev` independently as [I know a few use cases](https://github.com/richardscarrott/react-snap-carousel/issues/16#issuecomment-1682983180) where you'd want to sometimes animate and other times jump instantly.

Something like this
```ts
type SnapCarouselGoToOptions = {
  readonly behavior: ScrollBehavior;
};

// ...

const { scrollRef, prev, next, goTo } = useSnapCarousel();
return (
  <div style={styles.root}>
    <button onClick={() => goTo(0, { behavior: 'auto' })}>start</button>
    <button onClick={() => prev({ behavior: 'auto' })}>prev</button>
    <button onClick={() => next({ behavior: 'auto' })}>next</button>

    <ul style={styles.scroll} ref={scrollRef}>
      {items.map((item, i) =>
        renderItem({
          item,
          isSnapPoint: snapPointIndexes.has(i)
        })
      )}
    </ul>
  </div>
);
```

**Comment by richardscarrott** [comment]
Released in [0.4.0](https://github.com/richardscarrott/react-snap-carousel/tree/v0.4.0), thanks for the contribution @pantajoe 🚀
